### ISSUE TITLE
Getting Sentence level log probabilities using this model

### ISSUE THREAD

**Issue by Shashi456** [huggingface/pytorch-pretrained-BERT#477]
So I was trying to implement the SLOR score, using the transformer-XL, mostly avoiding training. But given the XL model and the sentence, how could i go about getting the sentence level log probability?

Attached is the formula I'm trying to implement.
![image](https://user-images.githubusercontent.com/18056781/56021217-81d6f700-5d26-11e9-8c46-9b88f0277398.png)

Thanks a lot for your help.

**Comment by thomwolf** [comment]
So p_M(S) is just the output of the model right?

For p_u(S), I think the easiest is probably to use the empirical probabilities.
`TransfoXLTokenizer` has a counter which store words' frequency [here](https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/pytorch_pretrained_bert/tokenization_transfo_xl.py#L98) so I would use and normalize this to get unconditional probabilities for each word and then compute SLOR.

**Comment by GregoireMialon** [comment]
> So p_M(S) is just the output of the model right?
> 
> For p_u(S), I think the easiest is probably to use the empirical probabilities.
> `TransfoXLTokenizer` has a counter to store words frequencies [here](https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/pytorch_pretrained_bert/tokenization_transfo_xl.py#L98) which should be populated in the "pretrained" tokenizer so I would use and normalize this to get unconditional probabilities for each word and then compute SLOR.

Is it still possible to get a populated counter ? For example, `tokenizer = TransfoXLTokenizer.from_pretrained('transfo-xl-wt103')` returns an empty counter.

**Comment by davebulaval** [comment]
In case someone found this, I found an implementation that seems legit here https://github.com/sgondala/GoogleConceptualCaptioning/commit/981e1b61ca5f84052f6237402319714d7fe70b80.
